client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -275,6 +277,9 @@ private boolean shouldAcquireSegment() throws ReinitializationRequiredException
             if (sync.getState().getNumberOfUnassignedSegments() == 0) {
                 return false;
             }
+            if (sync.getState().getCheckpointForReader(readerId) != null) {
```

Comment:
Here we are acquiring, do we need to be at a checkpoint to be able to acquire? I thought it was only needed for releasing a segment.